frontend/src/scenes/billing/billingLogic.ts

```diff
@@ -146,6 +147,45 @@ export const billingLogic = kea<billingLogicType>([
         billingAlert: [
             (s) => [s.billing, s.preflight, s.projectedTotalAmountUsd],
             (billing, preflight, projectedTotalAmountUsd): BillingAlertConfig | undefined => {
+                if (router.values.searchParams['products']) {
+                    let upgradedProducts = router.values.searchParams['products'].split(',')
+                    upgradedProducts = billing?.products.filter((product) => upgradedProducts.includes(product.type))
+                    const alerts: BillingAlertConfig[] = []
+                    upgradedProducts?.forEach((product: BillingProductV2Type) => {
+                        const currentPlan = product.plans.find((plan) => plan.current_plan)
+                        if (currentPlan?.initial_billing_limit) {
+                            alerts.push({
+                                status: 'warning',
+                                title: 'Billing Limit Automatically Applied',
+                                message: `To protect your and our costs, we've automatically applied a $${currentPlan?.initial_billing_limit} billing limit for ${product.name}.`,
+                                action: {
+                                    onClick: () => {
+                                        const element: HTMLElement = document.body.querySelector(
+                                            `[data-attr="billing-limit-input-${product.type.replace(
+                                                '_',
+                                                '-'
+                                            )}"] .text-link`
+                                        ) as HTMLElement
+                                        element?.scrollIntoView({ block: 'center', behavior: 'smooth' })
+                                        element?.click()
+                                        setTimeout(() => {
+                                            const inputElement: HTMLElement = document.body.querySelector(
+                                                `[data-attr="billing-limit-input-${product.type.replace(
+                                                    '_',
+                                                    '-'
+                                                )}"] input`
+                                            ) as HTMLElement
```

Comment:
this feels a little un-react. I'd probably:

- Use `billingProductLogic.setIsEditingBillingLimit` to show the input (this logic is keyed so you will only set it for the _one_ billingProductLogic that you are interested in)
- Put a ref on each billing limit input
- Use the the ref's `scrollIntoView` method to scroll to it, eg https://www.carlrippon.com/scrolling-a-react-element-into-view/
- Use the ref to focus the input

That way we keep it within react methods and don't directly manipulate the dom